benchmarking/evaluations/graphing/line_graph.py

```diff
@@ -40,13 +41,36 @@ def line_graph(graph_data: LineGraphMetadata):
         (0, (3, 1, 3, 1, 1, 1, 1, 1)),  # dashdash dotdotted
     ]
     markers: List[str] = ["o", "v", "s", "d", "x", "P", "*"]
+    curr_data: GraphData
```

Comment:
also an interesting pattern, as a `nit;` perhaps consider using `typing.cast` on it within the loop, but tbh this is fine